clmm/galaxycluster.py

```diff
@@ -181,28 +209,133 @@ def compute_tangential_and_cross_components(
+        pzbins : string
+            column name : redshift axis on which the individual photoz pdf is tabulated
+        shape_component1: string
+            column name : The measured shear (or reduced shear or ellipticity)
+            of the source galaxies
+        shape_component2: array
+            column name : The measured shear (or reduced shear or ellipticity)
+            of the source galaxies
+        shape_component1_err: array
+            column name : The measurement error on the 1st-component of ellipticity
+            of the source galaxies
+        shape_component2_err: array
+            column name : The measurement error on the 2nd-component of ellipticity
+            of the source galaxies
+        use_photoz : boolean
+            True for computing photometric weights
+        use_shape_noise : boolean
+            True for considering shapenoise in the weight computation
+        use_shape_error : boolean
+            True for considering measured shape error in the weight computation
+        weight_name : string
+            Name of the new column for the weak lensing weights in the galcat table
+        p_background_name : string
+            Name of the new column for the background probability in the galcat table
+        recompute_p_background: boolean
+            Forces re-computation of p_background if already in catalog.
+        is_deltasigma: boolean
+            Indicates whether it is the excess surface density or the tangential shear
+        add : boolean
+            If True, add weight column to the galcat table
+
+        Returns
+        -------
+        w_ls: array
+            the individual lens source pair weights
+        p_background : array
+            the probability for being a background galaxy
+        """
+        # input cols
+        required_cols = ['shape_component1', 'shape_component2']
+        if use_photoz:
+            required_cols += ['pzpdf', 'pzbins']
+        elif is_deltasigma:
+            required_cols += ['z_source']
+        if use_shape_error:
+            required_cols += ['shape_component1_err', 'shape_component2_err']
+        cols = self._get_input_galdata(locals(), required_cols)
+        # handles p_background
+        if p_background_name not in self.galcat.columns or recompute_p_background:
+            print(f'(re)computing {p_background_name}')
+            self.compute_background_probability(
+                z_source=z_source, pzpdf=pzpdf, pzbins=pzbins,
+                use_photoz=use_photoz, p_background_name=p_background_name)
+        cols['p_background'] = self.galcat[p_background_name]
+        # computes weights
+        w_ls = compute_galaxy_weights(
+            self.z, cosmo, use_shape_noise=use_shape_noise,
+            is_deltasigma=is_deltasigma, validate_input=self.validate_input,
+            **cols)
+        if add:
+            self.galcat[weight_name] = w_ls
+        return w_ls
+
     def make_radial_profile(self,
```

Comment:
So we now have the options to compute weights, but I do not see this propagated as an option to perfom a weighted average in `make_radial_profile`. @payerne, I think you did that in the issue/127 (cluster ensemble) branch (see `make_radial_profile` in `dataops/__init__.py in branch 127`). Just to clarify - do we agree to postpone this when dealing with `issue/127/...`. @payerne, @m-aguena ?